### ISSUE TITLE
False positive "@var is immediately overridden" when dealing with arrays 

### ISSUE THREAD

**Issue by olegtsvetkov** [kalessil/phpinspectionsea#672]
While working on some project, I got a false-positive warning about variable is being immediately overridden, but it's not.

Here is the code to reproduce the warning:
```php
$i = 0;
$array = [];

$array[$i++] = 1;
$array[$i++] = 2;
```

_I'm using PHP Inspections (EA Extended) 2.3.13.1 under PHPStorm 2017.3._

**Comment by kalessil** [comment]
Thank you for reporting @olegtsvetkov, will be fixed in the next release.

**Comment by kalessil** [comment]
Fixed
